src/Common/src/Interop/Unix/System.Net.Security.Native/Interop.NetSecurityNative.cs

```diff
@@ -0,0 +1,172 @@
+            bool isNtlm,
+            SafeGssNameHandle targetName,
+            uint reqFlags,
+            byte[] inputBytes,
+            int inputLength,
+            ref GssBuffer token,
+            out uint retFlags);
+
+        [DllImport(Interop.Libraries.NetSecurityNative, EntryPoint="NetSecurityNative_AcceptSecContext")]
+        internal static extern Status AcceptSecContext(
+            out Status minorStatus,
+            ref SafeGssContextHandle acceptContextHandle,
+            byte[] inputBytes,
+            int inputLength,
+            ref GssBuffer token);
+
+        [DllImport(Interop.Libraries.NetSecurityNative, EntryPoint="NetSecurityNative_DeleteSecContext")]
+        internal static extern Status DeleteSecContext(
+            out Status minorStatus,
+            ref IntPtr contextHandle);
+
+        [DllImport(Interop.Libraries.NetSecurityNative, EntryPoint="NetSecurityNative_Wrap")]
+        private static extern Status Wrap(
+            out Status minorStatus,
+            SafeGssContextHandle contextHandle,
+            bool isEncrypt,
+            byte[] inputBytes,
+            int offset,
+            int count,
+            ref GssBuffer outBuffer);
+
+        [DllImport(Interop.Libraries.NetSecurityNative, EntryPoint="NetSecurityNative_Unwrap")]
+        private static extern Status Unwrap(
+            out Status minorStatus,
+            SafeGssContextHandle contextHandle,
+            byte[] inputBytes,
+            int offset,
+            int count,
+            ref GssBuffer outBuffer);
+
+        internal static Status WrapBuffer(
+            out Status minorStatus,
+            SafeGssContextHandle contextHandle,
+            bool isEncrypt,
+            byte[] inputBytes,
+            int offset,
+            int count,
+            ref GssBuffer outBuffer)
+        {
+            Debug.Assert(inputBytes != null, "inputBytes must be valid value");
+            Debug.Assert(offset >= 0 && offset <= inputBytes.Length, "offset must be valid");
+            Debug.Assert( count >= 0 && count <= (inputBytes.Length - offset), "count must be valid");
+            Debug.Assert(count + offset <= inputBytes.Length, "offset and count must be valid");
+
+            return Wrap(out minorStatus, contextHandle, isEncrypt, inputBytes, offset, count, ref outBuffer);
+        }
+
+        internal static Status UnwrapBuffer(
+            out Status minorStatus,
+            SafeGssContextHandle contextHandle,
+            byte[] inputBytes,
+            int offset,
+            int count,
+            ref GssBuffer outBuffer)
+        {
+            Debug.Assert(inputBytes != null, "inputBytes must be valid value");
+            Debug.Assert(offset >= 0 && offset <= inputBytes.Length, "offset must be valid");
+            Debug.Assert( count >=0 && count <= inputBytes.Length, "count must be valid");
+            Debug.Assert(count + offset <= inputBytes.Length, "offset and count must be valid");
```

Comment:
In `WrapBuffer` this line was redundant; here it isn't (because the previous assert doesn't look at `offset`).  Since `count + offset` can overflow, but `inputBytes.Length - offset` can't over/underflow (because `offset` was already validated as being between `0` and `inputBytes.Length`) this assert is bad, and needs to be written as

```C#
Debug.Assert(count <= inputBytes.Length - offset, optionalMsg);
```

They're mathematically the same, but not within a bounds check.